subcommands/keys/tuf_copy_targets.go

```diff
@@ -13,14 +13,20 @@ import (
 
 func init() {
 	copy := &cobra.Command{
-		Use:   "copy-targets <offline key archive> <new targets archive>",
-		Short: "Copy the target signing credentials from the offline key archive",
-		Run:   doCopyTargets,
-		Args:  cobra.ExactArgs(2),
+		Use:    "copy-targets <offline key archive> <new targets archive>",
+		Short:  "Copy the target signing credentials from the offline key archive",
+		Hidden: true,
+		Run:    doCopyTargets,
+		Args:   cobra.ExactArgs(2),
 		Long: `This command extracts the target signing credentials required for initializing 
 waves into a new tarball so that the offline key archive isn't required for
 rolling out production updates. This should be run after each target key
 rotation and distributed to the operator in charge of production OTAs.`,
+		Deprecated: `it will be removed in the future.
+Please, use a more secure way to rotate the targets key into a separate file:
+	fioctl keys tuf rotate-offline-key --role=targets
+		--creds=<root-creds.tar.gz> --targets-creds=<targets-creds.tar.gz>
+`,
```

Comment:
but what if i want to export the target key w/o having to do a rotation?

Reply:
We should be very careful about all the _what if customer wants_ statements, especially in security related topics like this.

IMHO the most secure way to work with TUF keys is to keep the root and targets private keys separately. This becomes especially relevant when we start looking at the HSM support. With the HSM enabled the factory _root_ admin might even have no access to the _targets_ key, if that responsibility is delegated to a different person. That person would generate the _targets_ key on their HSM and let the _root_ admin sign a new _root_ with a new targets public part.

As you see, above there is no any `creds.tar.gz` at all. But even without the HSM that workflow can be generated if we keep the root and targets keys in two separate files.

Now back to you question. We will always have a group of customers want to take it easy and don't bother with security. For them, it might look like a quick start to keep root and targets keys in one tarball, so I keep that functionality. But, there should be no _today I want this, tomorrow I want that_ with regards to security. Even if the customer wants to switch from keeping-it-all-in-one-file to a more secure way to keep the targets key separately - they should be forced to generate a new targets key. I would allow nothing like _keep it here and keep it there_; redundancy is better achieved by different means.

Reply:
@doanac the above paragraph looks to contain _strong opinions_, however, regarding the `copy-targets` I don't really have _that strong opinion_. So, if you have any reasonable argument why we _do need_ it - I'm fine, and will simply move it under the `fioctl keys tuf` section.

When I think about e.g. the HSM support, the customer will need to somehow migrate existing TUF keys into the HSM. Depending onto the HSM model, there are two options:
- if the HSM supports inserting the crypto key into it - the `copy-targets` might come handy.
- if it doesn't - we'll have to generate a new key anyway.

IMHO, when we start talking about HSMs, the private key should never cross its boundary (i.e. no in, no out); at least that's what was the requirement back when I was working on FIPS-140-2 compliance. Thus, (again IMHO) we should generate the new keys when migrating the customers to HSM anyway. Other than that, I simply see no use for `copy-targets`. So it's like _less commands means less maintenance_ thing for me.

Reply:
maybe just make the statement less opinionated then. Rather than "Please, use a more secure way to rotate" just say something like "Please do it this way".

Reply:
changed in force-push to be less opinionated